probe/kubernetes/logreadcloser.go

```diff
@@ -0,0 +1,166 @@
+package kubernetes
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"math"
+)
+
+type logReadCloser struct {
+	labels       []string
+	labelLength  int
+	readClosers  []io.ReadCloser
+	eof          []bool
+	buffer       bytes.Buffer
+	dataChannel  chan []byte
+	stopChannels []chan struct{}
+	eofChannel   chan int
+}
+
+// NewLogReadCloser reads from multiple io.ReadCloser, where data is available,
+// and annotates each line with the reader's label
+func NewLogReadCloser(readClosersWithLabel map[io.ReadCloser]string) io.ReadCloser {
+	stopChannels := make([]chan struct{}, len(readClosersWithLabel))
```

Comment:
could also use a `sync.WaitGroup` for this (has the advantage of shutting down the readClosers in parallel)

Reply:
Thanks, as an extram it seems like I also won't need the `stopChannels` anymore.